### ISSUE TITLE
Try using Go-git

### ISSUE THREAD

**Issue by gregorriegler** [remotemobprogramming/mob#448]
So I found this golang based git implementation https://github.com/go-git/go-git

I think that if we would use it, it could have several advantages:

- We could use an in-memory filesystem (I think go-git even has one) that would make our tests very fast
- We would not have to run git as an external executable anymore which might help with the Virus detections
- It would work without git (which is probably neglible)

What do you think? @simonharrer @hollesse

**Comment by simonharrer** [comment]
I understand why it might look appealing. But I would not go down that road. The only real benefit of it for the users would be that we might have less issues with virus scanners. Other benefits are for dev only and not a real issue, but it requires a lot of work. And I am pretty sure when we use it other stuff will pop up that might impact users: how complete is git go? what will be executed differently for which we do not have tests?

**Comment by hollesse** [comment]
I also think that it is not a good idea. To be honest I don't think that it will even fix the issue with the virus scanners. As we used git since the beginning and it never was a problem. I had a look at git go and it is not complete and I found some things we use which are not supported. As a conclusion I see a lot of effort, very little benefits for the user and even things we would need to do differently. Therefore I would stay with using the installed git version on the users computer.
